code/datums/components/mood.dm

```diff
@@ -301,5 +302,38 @@
 		if(ETHEREAL_CHARGE_ALMOSTFULL to ETHEREAL_CHARGE_FULL)
 			add_event(null, "charge", /datum/mood_event/charged)
 
+
+/datum/component/mood/proc/HandleHygiene(mob/living/carbon/human/H)
+	switch(H.hygiene)
+		if(0 to HYGIENE_LEVEL_DIRTY)
+			if(has_trait(TRAIT_NEAT))
+				add_event(null, "neat", /datum/mood_event/dirty)
+			if(has_trait(TRAIT_NEET))
+				add_event(null, "NEET", /datum/mood_event/happy_neet)
+			HygieneMiasma(H)
+		if(HYGIENE_LEVEL_DIRTY to HYGIENE_LEVEL_NORMAL)
+			if(has_trait(TRAIT_NEAT))
+				clear_event(null, "neat")
+			if(has_trait(TRAIT_NEET))
+				clear_event(null, "NEET")
+		if(HYGIENE_LEVEL_NORMAL to HYGIENE_LEVEL_CLEAN)
+			if(has_trait(TRAIT_NEAT))
+				add_event(null, "neat", /datum/mood_event/neat)
+			if(has_trait(TRAIT_NEET))
+				clear_event(null, "NEET")
+
+/datum/component/mood/proc/HygieneMiasma(mob/living/carbon/human/H)
+	// Properly stored humans shouldn't create miasma
+	if(istype(H.loc, /obj/structure/closet/crate/coffin)|| istype(H.loc, /obj/structure/closet/body_bag) || istype(H.loc, /obj/structure/bodycontainer))
+		return
```

Comment:
Yes, I suppose that technically, people who haven't showered that are hanging out in COFFINS won't create bad smells.

But you could just make a subtype of the Rot component, which would save the repeated gas code.

Reply:
then id have to keep adding/killing the component which seems ass

Reply:
More or less ass than duplicating a bunch of code that doesn't need to be duplicated.

Reply:
not sure, the code isnt the same anyways.